### ISSUE TITLE
LMDB map resizing

### ISSUE THREAD

**Issue by who-biz** [who-biz/qwerty-lmdb#4]
Map resizing needs implemented.  Error is encountered at block height 57375 with the following message: 

```
ERROR   Error adding txpool tx blob to db transaction: MDB_MAP_FULL: Environment mapsize limit reached
```

**Comment by who-biz** [comment]
current error log:

```
 DEBUG   tx added: f3a3feb56dbc2488bac34c1c5cd10590a0800f046ae2028acd1de0eef9abc081
DEBUG   transaction 76e1b60534d58cb9220f416542e392f6f885e2a91d3b93661179d87e356cafb7 came in processObjects
DEBUG   tx added: 76e1b60534d58cb9220f416542e392f6f885e2a91d3b93661179d87e356cafb7
DEBUG   transaction f277a3013aa98ce2b382959dea254d7f41e1e78934724e55af8d205ea8cf5b47 came in processObjects
DEBUG   tx added: f277a3013aa98ce2b382959dea254d7f41e1e78934724e55af8d205ea8cf5b47
DEBUG   +++++ BLOCK SUCCESSFULLY ADDED
DEBUG   id:	a82054da1f86175f8b1f28f71d7ab675ce00814f17fdc4240ae55c7125bf3121
DEBUG   PoW:	0000000000000000000000000000000000000000000000000000000000000000
DEBUG   HEIGHT 112229, difficulty:	82422295
DEBUG   block reward: 284043.40710682, fee = 0.00000000, coinbase_blob_size: 434, cumulative size: 434, 0(0/0)ms
DEBUG   transaction 0db8c0c339e781663457b5a777427c5d62bee95d66f4a11377fc59256989e2d8 came in processObjects
ERROR   internal error: transaction already exists at inserting in memory pool: Error adding txpool tx blob to db transaction: MDB_MAP_FULL: Environment mapsize limit reached
TRACE   [213.136.89.252:8196 OUT] Exception in connectionHandler: Failed to open cursor: MDB_BAD_TXN: Transaction must abort, has a child, or is invalid
DEBUG   [213.136.89.252:8196 OUT] Interrupt connection
DEBUG   [213.136.89.252:8196 OUT] writeHandler finished
TRACE   [213.136.89.252:8196 OUT] CLOSE CONNECTION
ERROR   [80.211.204.60:8196 OUT] Failed to invoke COMMAND_HANDSHAKE, closing connection.
TRACE   Failed to HANDSHAKE with peer 80.211.204.60:8196
DEBUG   Selected peer: 7569724830599316913 163.172.8.20:8196 [white=1] last_seen: d0.h00.m00.s11
DEBUG   Connecting to 163.172.8.20:8196 (white=1, last_seen: d0.h00.m00.s11)...
DEBUG   Connection to 163.172.8.20:8196 failed: Failed to query m_blocks: MDB_BAD_TXN: Transaction must abort, has a child, or is invalid
DEBUG   Selected peer: 10660775559435072752 51.83.67.8:8196 [white=1] last_seen: d0.h00.m00.s38
DEBUG   Connecting to 51.83.67.8:8196 (white=1, last_seen: d0.h00.m00.s38)...
DEBUG   Connection to 51.83.67.8:8196 failed: Failed to query m_blocks: MDB_BAD_TXN: Transaction must abort, has a child, or is invalid
DEBUG   Selected peer: 14051819202797519902 95.216.206.245:8196 [white=1] last_seen: d0.h00.m19.s25
DEBUG   Connecting to 95.216.206.245:8196 (white=1, last_seen: d0.h00.m19.s25)...
DEBUG   Connection to 95.216.206.245:8196 failed: Failed to query m_blocks: MDB_BAD_TXN: Transaction must abort, has a child, or is invalid
DEBUG   Selected peer: 8113172305295515888 83.34.150.233:8196 [white=1] last_seen: d0.h00.m04.s45
DEBUG   Connecting to 83.34.150.233:8196 (white=1, last_seen: d0.h00.m04.s45)...
TRACE   [116.203.121.22:8196 OUT] NOTIFY_RESPONSE_CHAIN_ENTRY: m_block_ids.size()=10000, m_start_height=112198, m_total_height=456080
TRACE   [116.203.121.22:8196 OUT] Exception in connectionHandler: Failed to open cursor: MDB_BAD_TXN: Transaction must abort, has a child, or is invalid
```

**Comment by who-biz** [comment]
hangs at block 112229, improvement over previous (block 51375)

**Comment by exploshot** [comment]
you run into the same problem as me. Try to reduce the initial mapsize to maybe 64mb and extra size to 32mb.

try to take a look to https://github.com/exploshot/qwc-test/commit/ca7fedccd7591704897e8a6ea6f9b965510e0831 
the keyfile is LmDBWrapper.cpp

**Comment by who-biz** [comment]
@exploshot thanks. i'll take a look

**Comment by who-biz** [comment]
Sync now progresses to block 160,000 or so.

**Comment by who-biz** [comment]
These issues appear fixed as of commit 8860041a813b3036c5688fd1786be00facb7c53d.  Note that changing default map size and extra size did not appear to properly address the issue.  However, after exposing `BlockchainLMDB`s function `do_resize` to the parent class `BlockchainDB`, and making accessible to `Blockchain`, things seem to be moving along properly.

I tried various values for mapsize prior to this, ranging from 64MB to 1GB, none of which changed outcome until `m_db->do_resize` was able to be called from `Blockchain` class.

This means that many of the existing functions in `BlockchainLMDB` related to resizing, as well as DB-specific functions in `Blockchain` are unused, and can be eliminated to reduce needless complexity.

**Comment by who-biz** [comment]
Closing due to above. Will re-open if issues arise
